ico/git.py

```diff
@@ -0,0 +1,41 @@
+from git import Repo
+import os
+import time
+
+join = os.path.join
+repo = Repo(os.getcwd())
+assert not repo.bare
+
+repo.config_reader()
+
+def git_sanitycheck():
+    git_ismaster()
+    git_isold()
+    git_isdirty()
+
+def git_isdirty():
+    if repo.is_dirty():
+        raise RuntimeError("The repository is not committed, won't continue. Please commit.")
+
+    return
+
+def git_ismaster():
+    # User can override git_isold checking for a week
+    if ((float(os.getenv("ICO_DEPLOY_ANYWAY", 0)) + 604800) > time.time()):
```

Comment:
As @ztane mentioned, you don't have to use that many parentheses in Python `if`, here it's perfectly fine to do this:
```
if float(os.getenv("ICO_DEPLOY_ANYWAY", 0)) + 604800 > time.time():
```

And if it's me, I would probably move `float(os.getenv("ICO_DEPLOY_ANYWAY", 0)) + 604800` to a separate variable before this condition just to make it a little bit easier to understand, something like this:
```
deploy_anyway = os.getenv("ICO_DEPLOY_ANYWAY", 0)  # actually I don't know what this value, should it contain which type?
one_week = 60 * 60 * 24 * 7
now = time.time()

if deploy_anyway + one_week > now:
```

Btw why is this time check done in this function and not in `git_isold()`? It doesn't seem to relate to the current function that much.